internal/utils_test.go

```diff
@@ -30,6 +30,96 @@ func assertReflectEqual(t *testing.T, a interface{}, b interface{}) {
 	}
 }
 
+func assertSelectorEqual(t *testing.T, a map[string]interface{}, b types.NamespaceSelector) {
+	t.Helper()
+
+	if !compareSelectors(a, b) {
+		t.Fatalf("%s != %s", a, b)
+	}
+}
+
+func compareStringArrays(a []interface{}, b []string) bool {
+	// Account for when b is []string(nil)
+	if len(a) == 0 && len(b) == 0 {
+		return true
+	}
+
+	// Create a string array from []interface{}
+	aTyped := make([]string, len(a))
+	for i, val := range a {
+		aTyped[i] = val.(string)
+	}
+
+	return reflect.DeepEqual(aTyped, b)
+}
+
+func compareSelectors(a map[string]interface{}, b types.NamespaceSelector) bool {
+	if includeA, ok := a["include"].([]interface{}); ok {
+		if !compareStringArrays(includeA, b.Include) {
+			return false
+		}
+	} else if len(b.Include) != 0 {
+		return false
+	}
+
+	if excludeA, ok := a["exclude"].([]interface{}); ok {
+		if !compareStringArrays(excludeA, b.Exclude) {
+			return false
+		}
+	} else if len(b.Exclude) != 0 {
+		return false
+	}
+
+	if matchLabelsA, ok := a["matchLabels"].(map[string]string); ok {
+		if !reflect.DeepEqual(matchLabelsA, b.MatchLabels) {
+			return false
+		}
+	} else if matchLabelsA != nil && b.MatchLabels != nil {
+		return false
+	}
+
+	if matchExpressionsA, ok := a["matchExpressions"].([]interface{}); ok {
+		if a["matchExpressions"] != b.MatchExpressions {
+			if b.MatchExpressions != nil {
```

Comment:
This is very indented, could you please remove one layer by doing this?
```suggestion
			if b.MatchExpressions == nil {
				return false
			}
```